### PR TITLE
WIP: Multiple Node Versions in Single Workspace

### PR DESCRIPTION
**Pull request by parisholley**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Issue Number: #434

## What is the new behavior?
User can pass a custom name to node_repositories (allowing multiple node versions at the same time), then the other rules in the project allow setting a customer "node_repository" to point to the custom version.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Submitting this pull as a PR as i'm not all that familiar with bazel and want to get some thought before going all the way down the path of supporting dynamic @nodejs alias names, given there are a quite a few places where the alias is hardedcode and no ctx is available (so some deeper refactoring would be needed).

### CODE REVIEW

**Comment by parisholley** [comment]
@googlebot I signed it!

File: internal/node/node_repositories.bzl [hunk 1]
```diff
@@ -724,10 +724,16 @@ def node_repositories(**kwargs):
         minimum_bazel_version = "2.1.0",
     )
 
+    # allow user to set a custom name (eg: node11, node12) to support multiple versions
+    name = kwargs.get('name', 'nodejs')
```
Comment (alexeagle) [inline internal/node/node_repositories.bzl:728]: you can do this in one step - pop returns the element

**Comment by alexeagle** [comment]
hey @parisholley any update on this one?

**Comment by parisholley** [comment]
i decided to just upgrade everything to latest node so i stopped work on it

**Pull request closed without merging**